gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -356,10 +358,16 @@ private ManagedChannel createSingleChannel() throws IOException {
 
     // Check DirectPath traffic.
     boolean useDirectPathXds = false;
-    if (isDirectPathEnabled() && isNonDefaultServiceAccountAllowed() && isOnComputeEngine()) {
-      CallCredentials callCreds = MoreCallCredentials.from(credentials);
-      ChannelCredentials channelCreds =
-          GoogleDefaultChannelCredentials.newBuilder().callCredentials(callCreds).build();
+    if (isDirectPathEnabled() && isCredentialDirectPathCompatible() && isOnComputeEngine()) {
+      ChannelCredentials channelCreds;
+      if (credentials == null) {
+        // GoogleDefaultChannelCredentials will use application default call credential.
+        channelCreds = GoogleDefaultChannelCredentials.newBuilder().build();
```

Comment:
I think we should skip the whole DirectPath feature if `credentials` is null like you originally proposed? This would still fail the emulator tests in Spanner if directPath is enabled.
For more context, `credentials` should never be null unless the `CredentialProvider` is explicitly overridden to a `NoCreentialsProvider` like [this](https://github.com/googleapis/sdk-platform-java/blob/main/showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/EchoClientTest.java#L107), which is probably what the Spanner tests did.

Reply:
The problem here is that if the use case mentioned by @olavloite is valid, i.e., if customers set `.setCallCredentialsProvider(...).setCredentials(NoCredentials.getInstance())`, should we support it? My understanding is yes, but I do not know who should be the approver.

About the spanner emulator test failure, I think the Spanner test should understand the environment and fix it in their repo.

Reply:
> About the spanner emulator test failure, I think the Spanner test should understand the environment and fix it in their repo.

Yes, we will take care of that (it's actually not the emulator tests, but some tests using a `MockSpannerService` implementation).

> I think we should skip the whole DirectPath feature if credentials is null like you originally proposed?

That sounds reasonable to me. I think that it would be reasonable to instruct users who want to use DirectPath in combination with `CallCredentials` per RPC, that they have to supply a default set of credentials as well.

Reply:
OK I reverted the change back. PTAL. @blakeli0 @olavloite